Setting up autocomplete in Form

AaronCoplan opened the pull request:
This PR replaces #368, which I have closed per recommendations on #289.  I only added the auto complete property to **Form**, I thought it might be best to implement field specific auto complete in a future PR, but I am open to discussion.

@GregoryHamel If you have any thoughts / review that would be awesome!

AaronCoplan commented:
@GregoryHamel  Any thoughts on this, or good to merge?

GregoryHamel commented on src/components/Form/Form.react.js:
```diff
@@ -45,14 +46,20 @@ function Form({
   onSubmit,
+  autocomplete,
 }: Props): React.Node {
+  // default for autocomplete is "off" if not passed in explicitly
```
Maybe use `defaultProps` instead? see [Flow's defaultProps](https://flow.org/en/docs/react/components/#toc-using-default-props-for-functional-components)

```jsx
defaultProps = {
    autocomplete: "off";
}
```

GregoryHamel commented on src/components/Form/Form.react.js:
```diff
@@ -37,6 +37,7 @@ export type Props = {|
   +className?: string,
   +action?: string,
   +method?: string,
+  +autocomplete?: "on" | "off",
```
If there is a `defaultProps` for `autocomplete` it should not be optional, what do you think?

GregoryHamel commented on src/components/Form/Form.react.js:
```diff
@@ -45,14 +46,20 @@ function Form({
+  autocomplete,
 }: Props): React.Node {
+  // default for autocomplete is "off" if not passed in explicitly
+  if (!autocomplete) {
```
Maybe use `defaultProps` instead? see [Flow's defaultProps](https://flow.org/en/docs/react/components/#toc-using-default-props-for-functional-components)

```jsx
defaultProps = {
    autocomplete: "off";
}
```

GregoryHamel replied:
The `Table` component uses it as a out-of-function definition like so:

```jsx
Table.defaultProps = {
  cards: false,
  striped: false,
  responsive: false,
};
```

AaronCoplan commented:
@GregoryHamel Good call with the defaultProps!  Implemented that and removed what I believe was an unnecessary call to `cn(...)` - I will go ahead and merge unless you have any other changes 👍

jonthomp commented:
:tada: This PR is included in version 1.25.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/tabler-react)
- [GitHub release](https://github.com/tabler/tabler-react/releases/tag/v1.25.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
